Source/WebCore/accessibility/AXTextMarker.cpp

```diff
@@ -880,10 +891,13 @@ AXTextMarker AXTextMarker::findMarker(AXDirection direction, AXTextUnit textUnit
                     resultMarker = AXTextMarker(*currentObject, start - objectBorder);
             } else {
                 int end = nextSentenceEndFromPosition(flattenedRuns, offset);
+                // If the current marker (this) is the same position from the end, start a new search from there.
                 if (direction == AXDirection::Previous && end <= objectBorder && end != -1)
                     resultMarker = AXTextMarker(*currentObject, end);
-                else if (direction == AXDirection::Next && end != -1 && end >= objectBorder)
+                else if (direction == AXDirection::Next && end != -1 && end >= objectBorder && end != (int)offset) {
```

Comment:
Is there a safer way to compare than C-casting an unsigned to an int?

Reply:
Thought about this too, and our offset would have to be greater than the max int size to overflow (which would mean an offset of over 2 billion). Happy to make this safer though. Do you have any suggestions?

Reply:
I believe there is a safe way of doing this in WTF but don't remember exactly from top of my head.

Reply:
Updated the PR to do the casting—using `Checked<Int>` to cast will automatically handle overflows.